cmake/sdp.cmake

```diff
@@ -4,6 +4,24 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+function(sdp_assembly hrt_srcs)
+  sdp_assembly_generate("${CMAKE_SOURCE_DIR}/${hrt_srcs}")
+  sdp_assembly_check("${CMAKE_SOURCE_DIR}/${hrt_srcs}")
+  sdp_assembly_prepare_install("${CMAKE_SOURCE_DIR}/${hrt_srcs}")
+  sdp_assembly_target_sources("${CMAKE_SOURCE_DIR}/${hrt_srcs}")
+
+  add_dependencies(app asm_check)
+endfunction()
+
+function(sdp_assembly_target_sources hrt_srcs)
+  foreach(hrt_src ${hrt_srcs})
+    get_filename_component(hrt_dir ${hrt_src} DIRECTORY)  # directory
+    get_filename_component(hrt_src_file ${hrt_src} NAME_WE)  # filename without extension
+    set(hrt_s_file "${hrt_dir}/${hrt_src_file}.s")
+    target_sources(app PRIVATE ${hrt_s_file})
```

Comment:
Functions should not be hardcoded to a given target / library but take this as a function argument.